### ISSUE TITLE
Accessibility improvements to the site nav

### ISSUE THREAD

**Issue by fstorr** [hackoregon/civic#1054]
@Dianna: you wanted one issue for everything, so here's (I think) everything!

If something in here doesn't make sense let me know.

 ## Remove `role="menu"` and `role="menuitem"` from pop-up elements

The [ARIA `menu` role](https://www.w3.org/TR/wai-aria-1.1/#menu) is one that gets used incorrectly a lot because the spec isn't as clear as it could be. Instead of writing a lot of detail here, [Adrian Roselli's Don't Use ARIA Menu Roles for Site Nav](http://adrianroselli.com/2017/10/dont-use-aria-menu-roles-for-site-nav.html) is a worthwhile read.

The key part of the <code class="language-markup">menu</code> spec is this:

> "A menu is often a list of common actions or functions that the user can invoke. **_The menu role is appropriate when a list of menu items is presented in a manner similar to a menu on a desktop application._**"

In other words: does it look like a menu bar similar to something you'd see in Word, or maybe even the Markdown formatting bar in GitHub?

In addition to that ARIA quote, the [ARIA In HTML spec for the <code class="language-markup">menuitem</code> role](https://w3c.github.io/html-aria/#index-aria-menuitem) states that menu items shouldn't have descendants that are interactive content (links, buttons, etc.). Our current code is this:

```html
<li class="…" tabindex="-1" role="menuitem" aria-disabled="false" style="…">
  <a class="…" href="/#work-with-us">Work With Us</a>
  <span class="…"></span>
</li>
```
A link is interactive content, which further supports not using <code class="language-markup">menu</code> and <code class="language-markup">menuitem</code>

As we remove the <code class="language-markup">menu</code> and <code class="language-markup">menuitem</code> attributes, we should also remove these attributes which are no longer relevant:

1. <code class="language-markup">tabindex="-1"</code>
2. <code class="language-markup">aria-disabled="false"</code>

What we should use instead is the [Disclosure Pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure)…

## Replace `aria-haspopup` with `aria-expanded` to implement the disclosure pattern

The ARIA 1.0 spec allowed `aria-has-popup="true"` to specify if an element controlled "pop-up" content, which would include the drop-down menu items we have in the Join The Movement and About menu items. The [ARIA 1.1 spec for `aria-haspopup`](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup) changed things so we can't do that any more—there are set values the attribute can take, and if <code class="language-markup">true</code> is used, it's now the equivalent of <code class="language-markup">menu</code>, which is what we're no longer using.

Unlike <code class="language-markup">aria-haspopup</code>, <code class="language-markup">aria-expanded</code> takes one of two values: `true` or `false`. If the pop-up content is showing (i.e., the user has clicked on JOIN THE MOVEMENT), `aria-expanded` would be `true`; when the pop-up content is hidden, it would be `false`.

There's more reading on this in [Adrian Roseill's Link + Disclosure Widget Navigation](http://adrianroselli.com/2019/06/link-disclosure-widget-navigation.html) (note the [USA.gov](https://www.usa.gov/) content).

 We should end up with this in a button's non-clicked state:

 ```html
<button aria-controls="…" aria-expanded="false" class="…" type="button">
   …
</button>
 ```

 ## Remove the <code class="language-markup">&lt;p&gt;</code> element inside the button

This issue doesn't affect the accessibility of this component, but conformant code is a good thing and we should strive to write it. The [HTML spec for the <code class="language-markup">&lt;button&gt;</code> element](https://www.w3.org/TR/html53/sec-forms.html#the-button-element) allows [phrasing content](https://www.w3.org/TR/html53/dom.html#phrasing-content-2), whereas the <code class="language-markup">&lt;p&gt;</code> element is [flow content](https://www.w3.org/TR/html53/dom.html#flow-content). If we need to wrap the button's text in something, we should use the <code class="language-markup">&lt;span&gt;</code> element.

## Remove `role="tooltip"` from the pop-up menu items

The pop-up menu lists are wrapped in a <code class="language-markup">&lt;div&gt;</code> with a <code class="language-markup">role="tooltip"</code> attribute. The [ARIA `tooltip` role](https://www.w3.org/TR/wai-aria-1.1/#tooltip) is meant to be used for customized versions of the HTML `title` attribute. The spec describes it as <q>a contextual popup that displays a description for an element</q>, which isn't what a menu pop-up is. The [ARIA 1.1 Authoring Practices Tooltip Widget content](https://www.w3.org/TR/wai-aria-practices-1.1/#tooltip) adds that tooltips shouldn't contain any interactive content (links, buttons, etc.), which adds more weight to removing the attribute.

## Add keyboard support for closing "popped up" elements

[WCAG 2.1's Content On Hover Or Focus](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) requires a method to close pop-up content that can obscure content underneath it, which our menu items can do. The standard way to implement this is to use the <kbd>Escape</kbd> key to close the pop-up content. [Knowbility's article on Content On Hover Or Focus](https://knowbility.org/blog/2018/WCAG21-1413ContentHoverFocus/) is a good read.

## The carat on menu items with pop-up content is almost invisible in Windows high-contrast mode

Windows has a high-contrast mode that can be toggled on and off using <kbd><kbd>Left shift</kbd> + <kbd>Left Alt</kbd> + <kbd>Print Screen</kbd></kbd>. When high-contrast mode is enabled, the carat is almost invisible against the high-contrast mode's black background. I created a [high-contrast-mode test page](https://fstorr.github.io/hack-oregon-civic-components/components/svg/img-with-svg-currentcolor/) for this and the following screenshots are from that.

This is how our carats currently look:

![screenshot of a button in high-contrast mode. The button has a white border and white text and a barely-visible downwards-facing carat to the right of the text](https://user-images.githubusercontent.com/169471/65455768-c5b54880-ddfc-11e9-9ab4-08219420a116.png)

We could update the SVG using `currentColor` and still use the <code class="language-markup">&lt;img&gt;</code> element to put it on the page, but the carat's color wouldn't reflect Windows' styling of the element the SVG is in. For example:

![A screenshot of a link and a button in Windows high-contrast mode. The link is styled with yellow text and the button with white text. In both instances, the carat is white.](https://user-images.githubusercontent.com/169471/65456219-b5ea3400-ddfd-11e9-9f43-08d6e9b7c94a.png)

A better approach is to embed the SVG directly in the HTML so that the `currentColor` changes depending on the element the SVG is inside of. For example:

![A screenshot of a link and a button in Windows high-contrast mode. The link is styled with yellow text and the carat is the same color, and the button is styled with white text and has a white carat.](https://user-images.githubusercontent.com/169471/65456354-f8137580-ddfd-11e9-9752-b3e9b566fa11.png)

### Updating the carat's SVG

Our current SVG is this:

```xml
<svg xmlns="http://www.w3.org/2000/svg" width="29" height="10" viewBox="0 0 29 10" fill="none">
  <g clip-path="url(#clip0)">
    <path d="M0.868164 0.619629L15.7106 9.10491" stroke="#201024" stroke-width="2"/>
    <path d="M14.1299 8.98895L28.9723 0.503672" stroke="#201024" stroke-width="2"/>
  </g>
  <defs>
    <clipPath id="clip0">
      <rect width="29" height="10" fill="white"/>
    </clipPath>
  </defs>
</svg>
```

If we embed the SVG directly in the HTML, we can make the following changes:

1. remove the <code class="language-markup">xmlns</code> attribute.
2. hide this purely-decorative element from assistive technology using <code class="language-markup">aria-hidden="true"</code> .

We also need to update the SVG to work with Windows high-contrast mode by:

1. Updating the way the <code class="language-markup">&lt;path&gt;</code> and <code class="language-markup">&lt;g&gt;</code> elements are coded.
2. Removing the white background so it doesn't clash with the dynamically-created path colors.

This is our final code:

```xml
<svg aria-hidden="true" fill="none" height="10" viewBox="0 0 29 10" width="29">
  <g style="color:#201024">
    <path d="M0.868164 0.619629L15.7106 9.10491" stroke="currentColor" stroke-width="2"></path>
    <path d="M14.1299 8.98895L28.9723 0.503672" stroke="currentColor" stroke-width="2"></path>
  </g>
</svg>
```

## Summing up

If we implement these changes, we should end up with this:

```html
<nav aria-label="Site">
  <ul>
    <li>
      <a href="…">Explore Civic</a>
    </li>
    <li>
      <button aria-expanded="false" aria-owns="join-popup" type="button">
        <span>Join The Movement</span>
        <svg aria-hidden="true" fill="none" height="10" viewBox="0 0 29 10" width="29">
          <g style="color:#201024">
            <path d="M0.868164 0.619629L15.7106 9.10491" stroke="currentColor" stroke-width="2"></path>
            <path d="M14.1299 8.98895L28.9723 0.503672" stroke="currentColor" stroke-width="2"></path>
          </g>
        </svg>
      </button>
      <ul hidden id="join-popup">
        <li>
          <a href="…">Work with us</a>
        </li>
        <li>
          <a href="…">Become a contributor</a>
        </li>
        …
      </ul>
    </li>
    …
  </ul>
</nav>
```
